sdk/eventhubs/azure-messaging-eventhubs/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,269 @@
+
+### Partition ownership changes a lot
+
+This can occur in situations where the network latency is high between storage and the client and ownership is lost to another processor instance because the `PartitionOwnershipExpirationInterval` has elapsed.  Each load balancing interval, the EventProcessorClient looks for ownership records where the expiration interval has elapsed.  If it has, it believes no one is currently processing that partition, so it will take ownership of it and start processing events from the last checkpoint.  In this scenario, it is possible to process events more than once.  One way to mitigate this is to increase `LoadBalancingUpdateInterval` and increase `PartitionOwnershipExpirationInterval`.
+
+### "...current receiver 'nil' with epoch '0' is getting disconnected"
+
+This error is expected when load balancing occurs. Load balancing is indeed an ongoing process.  When using the BlobCheckpointStore with your consumer, every ~30 seconds (by default) the consumer will check to see which consumers have a claim for each partition, then run some logic to determine whether it needs to 'steal' a partition from another consumer.  The service side mechanism used to 'steal' partitions is [Epoch][Epoch].
+
+The full error message looks something like:
+
+> New receiver 'nil' with higher epoch of '0' is created hence current receiver 'nil' with epoch '0'
+> is getting disconnected. If you are recreating the receiver, make sure a higher epoch is used.
+> TrackingId:<GUID>, SystemTracker:<NAMESPACE>:eventhub:<EVENT_HUB_NAME>|<CONSUMER_GROUP>,
+> Timestamp:2022-01-01T12:00:00}"}
+
+### High CPU usage
+
+* Usually because too many partitions are owned.
+* Recommend no more than 3 partitions to every 1 CPU core; better to start with 1.5 partitions to each core and test increasing
+
+### Processor client stops receiving
+
+Often this is not enough information to determine why the exception occurred. For the team to determine the cause, we need to ask for the following information from the customer:
+
+* Event Hub environment
+  * How many partitions?
+* EventProcessorClient environment
+  * What are the specs of the machine(s) processing your Event Hub?
+  * How many instances are running?
+* Repro code and steps
+  * This is important as many times we cannot reproduce the issue in our own environment.
+* Logs. We need DEBUG logs, but if that is not possible, INFO at least. Error and warning level logs do not provide enough information. Timespan of at least +/- 10 minutes from when the issue occurred.
+
+### Migrate from legacy to new client library
+
+The migration guide is here [Migration Guide](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/eventhubs/azure-messaging-eventhubs/migration-guide.md).  This includes steps on how to migrate checkpoints.
```

Comment:
Might looks cleaner with a slight re-wording as the first sentence is:

`The migration guide is here Migration Guide.`

```suggestion
The [migration guide](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/eventhubs/azure-messaging-eventhubs/migration-guide.md) includes steps on how to migrate from the legacy client, and includes how to migrate checkpoints.
```

(Though this still needs a little wordsmithing)